test/plugins/test-trace-hapi.js

```diff
@@ -308,6 +308,41 @@ describe('hapi', function() {
           });
         });
       });
+
+      it('should end spans when client aborts request', function(done) {
```

Comment:
First off,  💪 on the work, really appreciate it!
Second should `res.end` not always be called or am I getting this wrong? (https://nodejs.org/api/http.html#http_response_end_data_encoding_callback)